scripts/speech_recognition/convert_to_tarred_audio_dataset.py

```diff
@@ -644,39 +601,69 @@ def _read_manifest(self, manifest_path: str, config: ASRTarredDatasetConfig):
 
         return entries, total_duration, filtered_entries, filtered_duration
 
-    def _write_to_tar(self, tar, audio_filepath: str, squashed_filename: str) -> None:
-        if (codec := self.config.force_codec) is None or audio_filepath.endswith(f".{codec}"):
-            # Add existing file without transcoding.
+    def _write_to_tar(
+        self, tar, audio_filepath: str, squashed_filename: str, duration: float = None, offset: float = 0
+    ) -> None:
+        codec = self.config.force_codec
+        to_transcode = not (codec is None or audio_filepath.endswith(f".{codec}"))
+        to_crop = not (duration is None and offset == 0)
+
+        if not to_crop and not to_transcode:
+            # Add existing file without transcoding, trimming, or re-encoding.
             tar.add(audio_filepath, arcname=squashed_filename)
-        else:
-            # Transcode to the desired format in-memory and add the result to the tar file.
-            audio, sampling_rate = soundfile.read(audio_filepath, dtype=np.float32)
-            encoded_audio = BytesIO()
+            return
+
+        # Standard processing: read, trim, and transcode the audio file
+        audio, sampling_rate = soundfile.read(audio_filepath, dtype=np.float32)
+
+        # Trim audio based on offset and duration.
+        start_sample = int(offset * sampling_rate)
+        end_sample = int((offset + duration) * sampling_rate) if duration else None
+        audio = audio[start_sample:end_sample]
```

Comment:
Use `frames` and `start` arguments to `soundfile.read` to avoid reading the entire audio into memory. It will drastically speed up the script for long-form audio with offsets. 

https://github.com/bastibe/python-soundfile/blob/5069b8cb828eb2bb26aba0620e1f7ea2ae37f189/soundfile.py#L253-L257